File: rhinocommon/snippets/py/ex_addsphere.py
import Rhino
import scriptcontext
import System.Guid

def AddSphere():
    center = Rhino.Geometry.Point3d(0, 0, 0)
    radius = 5.0
    sphere = Rhino.Geometry.Sphere(center, radius)
    if scriptcontext.doc.Objects.AddSphere(sphere)!=System.Guid.Empty:
        scriptcontext.doc.Views.Redraw()
        return Rhino.Commands.Result.Success
    return Rhino.Commands.Result.Failure


if __name__ == "__main__":
    AddSphere()

File: rhinocommon/snippets/from_rhinocommon/vbnet/ex_addsphere.vb
Partial Class Examples
  Public Shared Function AddSphere(ByVal doc As Rhino.RhinoDoc) As Rhino.Commands.Result
    Dim center As New Rhino.Geometry.Point3d(0, 0, 0)
    Const radius As Double = 5.0
    Dim sphere As New Rhino.Geometry.Sphere(center, radius)
    If doc.Objects.AddSphere(sphere) <> Guid.Empty Then
      doc.Views.Redraw()
      Return Rhino.Commands.Result.Success
    End If
    Return Rhino.Commands.Result.Failure
  End Function
End Class
